test/metabase/automagic_dashboards/core_test.clj

```diff
@@ -208,6 +209,402 @@
             (perms/grant-collection-readwrite-permissions! (perms-group/all-users) collection-id)
             (test-automagic-analysis (db/select-one Card :id card-id) 8)))))))
 
+(deftest field-matching-predicates-test
+  (testing "The field spec predicate should match fields by their fieldspec"
```

Comment:
This doesn't test the `(comp #{fieldspec} :name)` branch of `field-spec-pred`. (I actually think that's ok because it's weird that this matches on the semantic type OR the name if you give it a string)

Reply:
Added a test. It's even weirder than that because the fieldspec in this branch has to be named "ga:fieldname."